packages/react/src/components/Table/Table.tsx

```diff
@@ -0,0 +1,57 @@
+import { ReactElement, CSSProperties } from 'react';
+import '@compassion-gds/css/src/components/Table/table.css';
+
+export interface ColumnCell {
+  title: string;
+  key: string;
+  headerRenderFn?: (key: string | number, title: string) => ReactElement;
+  cellRenderFn?: (key: string, data: string | number | object) => ReactElement;
+}
+
+export interface TableProps {
+  columns: ColumnCell[],
+  rows: object[],
+  disabled?: boolean,
+  stickyHeader?: boolean,
+  containerStyle?: CSSProperties,
+  tableStyle?: CSSProperties,
+}
+
+const Table = ({ columns = [], rows = [], disabled = false, stickyHeader = false, containerStyle, tableStyle }: TableProps) => {
+  const sticky = stickyHeader ? 'sticky-header' : '';
+  const disabledTable = disabled ? 'disabled' : '';
+  const classNames = ['table', sticky, disabledTable].join(' ');
+
+  return (
+    <div className="table-container" style={containerStyle}>
+      <table className={classNames} style={tableStyle}>
+        <thead className="table-head">
+          <tr className="table-row">
+            {
+              columns.map((col, index) => (
+                col.headerRenderFn
+                  ? col.headerRenderFn(index, col.title)
+                  : <th key={index} className="table-cell">{col.title}</th>
+              ))
+            }
+          </tr>
+        </thead>
+        <tbody className="table-body">
+          {rows.map((row, rowIndex) => (
+            <tr key={rowIndex} className="table-row">
+              {
+                Object.keys(row).map((key, cellIndex) => (
+                  columns[cellIndex].cellRenderFn
+                    ? columns[cellIndex].cellRenderFn(key, row[key])
+                    : <td className="table-cell" key={cellIndex}>{row[key]}</td>
+                ))
```

Comment:
Code here expects **perfect data** and can lead to bugs when the `row` has more keys or less keys and not exactly as the number of columns 

This won't work for cases where I pass JSON and want to only display certain columns 
For example I want to display only columns A, B, C, while each row has 100 keys 

Or when some rows might be lacking keys, for example a row is missing key B - the table would render less cells the row and they won't match the columns (everything is shifted to the left)

A table should render a cell for each column, cells shouldn't depend on arbitrary row structure
Something like this:
```suggestion
                columns.map((col) => (
                  col.cellRenderFn
                    ? col.cellRenderFn(col, row)
                    : <td className="table-cell" key={col.key}>{row[col.key]}</td>
                ))
```

Note: we're passing the whole row/col to `col.cellRenderFn(col, row)`, the `cellRenderer` can decide to use `row[col.key]` or if it needs more values from the `row` to render the cell 
While `col` can carry data like cell alignment (so a row can align the same as the header col) or other metadata